unit_tests/tests/test_call.rs

```diff
@@ -0,0 +1,299 @@
+            message: _,
+            code: MaybeUnknownErrorCode::Known(StarknetError::ContractNotFound)
+        }))
+    );
+}
+
+/**
+ * Unit test for `starknet_call`
+ * 
+ * purpose: function request `name` to StarkGate ETH bridge contract
+ * fail case: invalid field element
+ */
+#[rstest]
+#[tokio::test]
+async fn fail_invalid_contract_entry_point_selector(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall {
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(),
+            entry_point_selector: FieldElement::ZERO,
+            calldata: vec![]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.err();
+
+    assert_matches!(
+        response_deoxys,
+        Some(ProviderError::StarknetError(StarknetErrorWithMessage {
+            message: _,
+            code: MaybeUnknownErrorCode::Known(StarknetError::ContractError)
+        }))
+    );
+}
+
+/**
+ * Unit test for `starknet_call`
+ * 
+ * purpose: function request `balanceOf` to StarkGate ETH bridge contract
+ * fail case: missing call data. This is different from solely *invalid* call data, as we will see shortly
+ */
+#[rstest]
+#[tokio::test]
+async fn fail_missing_contract_call_data(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall {
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(),
+            entry_point_selector: get_selector_from_name("balanceOf").unwrap(),
+            calldata: vec![]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.err();
+
+    assert_matches!(
+        response_deoxys,
+        Some(ProviderError::StarknetError(StarknetErrorWithMessage {
+            message: _,
+            code: MaybeUnknownErrorCode::Known(StarknetError::ContractError)
+        }))
+    );
+}
+
+/**
+ * Unit test for `starknet_call`
+ * 
+ * purpose: function request `balanceOf` to StarkGate ETH bridge contract
+ * fail case: invalid call data. This does not cause an error upon calling the contract but returns felt 0x0
+ */
+#[rstest]
+#[tokio::test]
+async fn fail_invalid_contract_call_data(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall {
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(),
+            entry_point_selector: get_selector_from_name("balanceOf").unwrap(),
+            calldata: vec![
+                FieldElement::ZERO
+            ]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Deoxys node");
+
+    assert_eq!(response_deoxys, vec![FieldElement::ZERO, FieldElement::ZERO]);
```

Comment:
What kind of response is awaited here on both sides?

Reply:
This is just the default error return value by the node in case of a call with invalid call data. A bit janky but this is just how it is implemented on the node side.